src/aditiGJUN20/SumOfDigits.java

```diff
@@ -0,0 +1,42 @@
+// Find sum of all digits in given string. (Consider consecutive digits till next character as single number)
+
+package aditiGJUN20;
+
+public class SumOfDigits {
+	int getSumofAllDigits(String str) {
+		// Condition for empty String
+		if (str.isEmpty()) {
+			System.out.println("Null String");
+			return 0;
+		}
+		
+		int digitSum = 0;
+		String temp = "0"; // Earlier tried with "" and null but had thrown exception so settled with 0 as
+							// doing integer sum.
+		for (int index = 0; index < str.length(); index++) {
+			char ch = str.charAt(index);
+			if (Character.isDigit(ch))
+				temp += ch;
+			else {
+				digitSum += Integer.parseInt(temp);
+				temp = "0";
+			}
+		}
+		int finalSum = digitSum + Integer.parseInt(temp);
+		if (finalSum == 0 && str.contains("0"))
```

Comment:
Here if the last digit is a number, then again one validation is required outside of the For loop:

if(temp.length() != 0) {
			sum += Integer.parseInt(temp);
		}
		return sum;

Reply:
@AmitaRJun20 
I was referring to this condition only which i'd missed. I've added it in code and removed the extra variable.
Thank you for detailed review.